### DISCUSSION TITLE
Is it possible to call a Hasura GraphQL REST endpoint from a Cron Event?

### DISCUSSION THREAD

**Discussion by patrickdevivo** [hasura/graphql-engine#7917] [General] [answered]
Hi all! I'm having trouble setting up the following:

1. I have a mutation defined in my GraphQL schema that takes no variables
2. I have a REST endpoint defined for that mutation (`POST` request, expects no params other than an authentication header)
3. I'd like a `Cron Trigger` to execute that mutation on an interval

I'm able to define the Cron Trigger, pointed at my REST endpoint with an empty (`{}`) payload and the correct headers to pass.

However, the invocations fail with a `400` and the following message:

```json
{
  "body": "{\"path\":\"$\",\"error\":\"Unexpected variable scheduled_time\",\"code\":\"bad-request\"}"
}
```

I believe this is because of the "double" `payload` in the actual JSON body sent to the REST endpoint:

```json
{
    "payload": {
        "scheduled_time": "2021-12-13T04:46:00Z",
        "payload": {},
        "name": "Cron Trigger Name",
        "id": "55eea74e-87ed-4672-ac5b-20a865ceb897"
    },
    "headers": [
        {
            "value": "application/json",
            "name": "Content-Type"
        },
        {
            "value": "hasura-graphql-engine/v2.0.10-cloud.1",
            "name": "User-Agent"
        }
    ],
    "version": "1"
}
```

I'm wondering if anyone has ideas on how to get around this?

**Comment by tirumaraiselvan** [comment, marked as answer]
Hi, this should be possible very soon with Schedule Trigger Transforms in which you can change the payload structure of a cron event to adhere to the the mutations payload. See similar issue here: https://github.com/hasura/graphql-engine/issues/5050

**Reply by patrickdevivo** [reply]
Great! Thank you for letting me know - do you know of a place I can track the work happening for `Schedule Trigger Transforms`? Would love to know when it becomes available. I wasn't able to find something like that in the linked/related issue(s). Thanks for the info!
